TCC.Lib/TarCompressCrypt.cs

```diff
@@ -170,23 +170,51 @@ public EncryptionKey(string key, string keyCrypted)
             return new EncryptionKey(key, keyCrypted);
         }
 
-        private static string CompressCommand(Block block, TccOption option, ExternalDependecies ext)
+        private static string CompressCommand(Block block, CompressOption option, ExternalDependecies ext)
         {
             string cmd;
             switch (option.PasswordOption.PasswordMode)
             {
                 case PasswordMode.None:
                     // tar -c C:\SourceFolder | lz4.exe -1 - compressed.tar.lz4
                     cmd = $"{ext.Tar().Escape()} -c {block.Source.Escape()}";
-                    cmd += $" | {ext.Lz4().Escape()} -1 -v - {block.DestinationArchive.Escape()}";
+
+                    switch (option.Algo)
+                    {
+                        case CompressionAlgo.Lz4:
+                            cmd += $" | {ext.Lz4().Escape()} -1 -v - {block.DestinationArchive.Escape()}";
+                            break;
+                        case CompressionAlgo.Brotli:
+                            cmd += $" | {ext.Brotli().Escape()} - -o {block.DestinationArchive.Escape()}";
+                            break;
+                        case CompressionAlgo.Zstd:
+                            cmd += $" | {ext.Zstd().Escape()} - -o {block.DestinationArchive.Escape()}";
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException(nameof(option.Algo), "Unknown PasswordMode");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The parameter name 'Algo' is not declared in the argument list. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)